### PR TITLE
New package: swan

### PR DESCRIPTION
**Pull request by lhxone**
Add builtin-package swan.

### CODE REVIEW

File: var/spack/repos/builtin/packages/swan/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+from spack import *
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/swan/package.py:6]: Please restore the standard Spack package copyright.

File: var/spack/repos/builtin/packages/swan/package.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+class Swan(MakefilePackage):
+    """SWAN is a third-generation wave model, developed at Delft 
+    University of Technology, that computes random, short-crested
+     wind-generated waves in coastal regions and inland waters. 
+    For more information about SWAN, see a short overview of model 
+    features. This list reflects on the scientific relevance of 
+    the development of SWAN."""
+
+    # FIXME: Add a proper url for your package's homepage here.
+    homepag="http://swanmodel.sourceforge.net/"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/swan/package.py]:
```suggestion
    homepage="http://swanmodel.sourceforge.net/"
```

File: var/spack/repos/builtin/packages/swan/package.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+from spack import *
+
+class Swan(MakefilePackage):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/swan/package.py:8]: You might want to consider leveraging the `SourceforgePackage` mix-in.  See https://spack.readthedocs.io/en/latest/_modules/spack/build_systems/sourceforge.html?highlight=SourceforgePackage.

There are currently 44 package that use it.  Examples include `octave-*` packages, `swig`, `tcl`, and `tk`.

File: var/spack/repos/builtin/packages/swan/package.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    """SWAN is a third-generation wave model, developed at Delft 
+    University of Technology, that computes random, short-crested
+     wind-generated waves in coastal regions and inland waters. 
+    For more information about SWAN, see a short overview of model 
+    features. This list reflects on the scientific relevance of 
+    the development of SWAN."""
+
+    # FIXME: Add a proper url for your package's homepage here.
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/swan/package.py]: Please remove this line.

File: var/spack/repos/builtin/packages/swan/package.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    homepag="http://swanmodel.sourceforge.net/"
+    url="https://cfhcable.dl.sourceforge.net/project/swanmodel/swan/41.31/swan4131.tar.gz"
+
+    # FIXME: Add a list of GitHub accounts to
+    # notify when the package is updated.
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/swan/package.py]: Please remove these lines.

File: var/spack/repos/builtin/packages/swan/package.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    depends_on('mpich', type = 'build')
+    depends_on('netcdf-fortran', type = ('build', 'run'))
+    depends_on('libfabric', type = ('run'))
+
+    def edit(self, spec, prefix):
+        env['FC']='gfortran'
+        m = FileFilter('platform.pl')
+        m.filter('F90_MPI = .*', 'F90_MPI = mpifort\\n";')
+        m.filter('NETCDFROOT =', 'NETCDFROOT = {0}' + spec['netcdf-fortran'].prefix))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/swan/package.py]:
```suggestion
        m.filter('NETCDFROOT =', 'NETCDFROOT = {0}' + spec['netcdf-fortran'].prefix)
```

File: var/spack/repos/builtin/packages/swan/package.py (hunk 1 above)
Comment (lhxone) [reply var/spack/repos/builtin/packages/swan/package.py:6]: fixed

File: var/spack/repos/builtin/packages/swan/package.py (hunk 5 above)
Comment (lhxone) [reply var/spack/repos/builtin/packages/swan/package.py]: removed

File: var/spack/repos/builtin/packages/swan/package.py (hunk 3 above)
Comment (lhxone) [reply var/spack/repos/builtin/packages/swan/package.py:8]: Thanks for you review. But I don't know much about `class SourceForge`. So I think it's better to keep it as `class Swan(MakefilePackage):`

**Comment by lhxone** [comment]
The above are all fixed, can I ask for a re-merge?

**Pull request closed without merging**